src/lib.rs

```diff
@@ -503,6 +547,59 @@ impl<'a, T> Iterator for WriteIter<'a, T> {
 
 
 /// Item of the streaming iterator.
+///
+/// [`Cursor`](struct.Cursor.html) and `CursorItem` are extremely useful
+/// when you need to iterate over [`Pointers`](struct.Pointer.html) in storage
+/// with ability to get other component in the same storage during iterating.
+///
+/// # Examples
+/// Unfortunately, you can't use common `for` loop,
+/// but you can use `while let` statement:
+///
+/// ```rust
+/// # let mut storage: froggy::Storage<i32> = froggy::Storage::new();
+/// let mut cursor = storage.cursor();
+/// while let Some(item) = cursor.next() {
+///    // ... your code
+/// }
+///
+/// ```
+/// While iterating, you can [`pin`](struct.CursorItem.html#method.pin) item
+/// with Pointer. Also you can [`look_ahead`](struct.CursorItem.html#method.look_ahead)
+/// and [`look_back`](struct.CursorItem.html#method.look_back) for other items
+/// in this storage.
+///
+/// ```rust
+/// # use froggy::WeakPointer;
+/// #[derive(Debug, PartialEq)]
+/// struct Node {
+///    pointer: Option<WeakPointer<Node>>,
+///    flag: bool
+/// }
+/// # fn try_main() -> Result<(), froggy::Error> {
+/// # let mut storage = froggy::Storage::new();
+/// let ptr1 = storage.create(Node { pointer: None, flag: false });
```

Comment:
Well, I thought about it. The problem is, `Cursor` is needed in rare situations, I have no idea about small and self-explanatory example. Do you have one?